mesop/dataclass_utils/dataclass_utils_test.py

```diff
@@ -94,6 +97,19 @@ class RootDataclass:
   assert instance.nested.val == "<default>"
 
 
+def test_dataclass_defaults_works_with_already_annotated_nested_class_requires_default():
+  @dataclass
+  class NestedDataclass:
+    val: str = "default"
+
+  @dataclass_with_defaults
+  class RootDataclass:
+    nested: NestedDataclass
+
+  instance = RootDataclass()
+  assert instance.nested.val == "default"
+
+
 def test_dataclass_defaults_works_with_already_annotated_nested_class_fails_without_default():
```

Comment:
[question] This is a pre-existing test name, but I'm a bit confused by the test name here. Is this testing something that is valid or that fails? Seems like something valid.

Reply:
it's basically testing that NestedDataclass which is already annotated with @dataclass fails because it doesn't have a default value.